src/index.ts

```diff
@@ -145,42 +145,35 @@ export class CruxClient {
     }
 
     public walletClientName: string;
-    protected _options: ICruxClientOptions;
-    protected _getEncryptionKey: () => string;
-    protected _keyPair?: blockstackService.IBitcoinKeyPair;
+    protected _getEncryptionKey: () => Promise<string>;
+    protected _keyPair?: string | blockstackService.IBitcoinKeyPair;
     protected _encryption: typeof encryption.Encryption;
     protected _nameService?: nameService.NameService;
     protected _payIDClaim?: PayIDClaim;
     protected _configService?: configurationService.ConfigurationService;
-    private _authenticated: boolean;
+    private _authenticated!: boolean;
     private initPromise: Promise<void>;
+    private _setKeyPairPromise: Promise<void>;
 
     constructor(options: ICruxClientOptions) {
-        this._options = Object.assign({}, options);
         // TODO: Need to validate options
 
-        if (this._options.getEncryptionKey) {
-            this._getEncryptionKey = this._options.getEncryptionKey;
+        if (options.getEncryptionKey) {
+            this._getEncryptionKey = options.getEncryptionKey;
         } else {
             const encryptionKeyConst = utils.getRandomHexString();
-            this._getEncryptionKey = (): string => encryptionKeyConst;
+            this._getEncryptionKey = async (): Promise<string> => encryptionKeyConst;
         }
         // log.debug(`Encryption key:`, this._getEncryptionKey())
 
-        // Setting up the default modules as fallbacks
-        if (this._options.privateKey) {
-            this._keyPair =  utils.getKeyPairFromPrivKey(this._options.privateKey);
-            this._authenticated = true;
-        } else {
-            this._authenticated = false;
-        }
+        this._encryption = options.encryption || encryption.Encryption;
+        this._nameService = options.nameService;
+        this.walletClientName = options.walletClientName;
 
-        this._encryption = this._options.encryption || encryption.Encryption;
-        this._nameService = this._options.nameService;
-        this.walletClientName = this._options.walletClientName;
+        this._setKeyPairPromise = this._setKeyPair(options.privateKey);
```

Comment:
are we cleaning privateKey from options once it's assigned to keyPair?

Reply:
removing post call to the setter method